### PR TITLE
Dont block user on mautic

### PR DESCRIPTION
**Pull request by sirpy**
# Description
during signup we block until data is updated in mautic/database etc this is not critical for user, so i've removed blocking

- dont block on user signup
- removed topwallet form signup (user now uses the faucet contract through app)
- removed temp mautic contact creation on verify email, no longer required since we send through SES

QA
- check user signup is working as expected
- new user doesnt have gas issue
- verify email step in signup(with passwordless) and when editing on profile works fine and also updates the user email on mautic correctly

### CODE REVIEW

File: src/server/utils/timeout.js [hunk 1]
```diff
@@ -22,3 +22,6 @@ export default async function requestTimeout(millis, timeoutReason = null) {
 
   await timeout(millis, errorMessage)
 }
+
+Promise.timeout = (promises, timeout, timeoutReason) =>
```
Comment (serdiukov-o-nordwhale) [inline src/server/utils/timeout.js:26]: 1. do not need to make it 'universal', let it just receives some promise. If someone would need pass array of promises, he would wrap it to Promise.all by himself
2. we probably may want to use it with retry(). So let pass not a promise itself but the async function returning promise
3. Modifying existing JS objects is a kind of the bad design. it should be used only if we want to polyfill some spec which doesn't exists on the JS engine we're using.

Following the pt 1-3 i suggest to usethe implementation below:

```javascript
export const runWithTimeout(asyncFn, millis, timeoutReason = null) =>
  Promise.race([asyncFn(), requestTimeout(millis, timeoutReason)])
```

**Review by serdiukov-o-nordwhale — CHANGES_REQUESTED** [review]

**Comment by serdiukov-o-nordwhale** [comment]
abouts pt2 - ok let's fix when we'll need that
but pt3 should be fixed as it's really bad design. Promise interface is strongly defined: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
There's no Promise.timeout, so we shouldn't polyfill that method. If you don't like runWithTimeout name - you could choose your own

**Comment by serdiukov-o-nordwhale** [comment]
@sirpy  how about pt3 ? would you fix it ? i really recommend to do that because changing generic interfaces is a bad idea

**Comment by sirpy** [comment]
you can modify it if you want.
i dont think it is that important, many libraries extend interfaces

**Comment by serdiukov-o-nordwhale** [comment]
> you can modify it if you want. i dont think it is that important, many libraries extend interfaces

yes, this was popular some time ago. But after HTML5 standard was released, the concept was changed to keep the existing interfaces.
From the all languages/frameworks/etc i met last time i saw 'extension' feature only in the Swift. But Swift doesn't just extends some class with some method. You have define protocol (it's like interface in Java) firstly then add extension for conform that protocol. So the interface segregation principle isn't breaking here.

**Review by serdiukov-o-nordwhale — APPROVED** [review]

File: src/server/storage/storageAPI.js [hunk 2]
```diff
@@ -193,12 +194,26 @@ const setup = (app: Router, gunPublic: StorageAPI, storage: StorageAPI) => {
...
+
+        const p6 = withTimeout(p5, 15000, 'updated trust indexes timeout').catch(e => {
           logger.error(e.message)
         })
+
         signUpPromises.push(p6)
 
-        await Promise.all(signUpPromises)
+        // dont await, if we failed to update its not critical for user.
+        withTimeout(signUpPromises, 30000, 'signup promises timeout')
```
Comment (sirpy) [inline src/server/storage/storageAPI.js:205]:
```suggestion
        withTimeout(Promise.all(signUpPromises), 30000, 'signup promises timeout')
```

**Comment by iLystopad** [comment]
@sirpy The user is not able to sign up with passwordless. 
{from: "SignupState"} "New user failure" "User email not verified!" E8: User email not verified! is displayed in the console after the user clicking on the "Let's start" button

Checked on the QA 1.23.1-1
Device: Desktop // Windows 10 // Chrome 88.0.4324.150

Attachment:

**Comment by iLystopad** [comment]
@sirpy 
"Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Bok%2C%20error%7D&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings" is displayed in the console after entering invalid verification code on email verification page

Checked on the QA 1.23.1-1
Device: Desktop // Windows 10 // Chrome 88.0.4324.150

Note: Also reproduced after editing email and entering invalid verification code on the Edit Profile page

Attachment:

**Comment by sirpy** [comment]
@iLystopad try now

**Comment by iLystopad** [comment]
@sirpy The issue with signup via passwordless is still reproduced. The issue with entering an invalid verification code is fixed.

The user is not able to sign up with passwordless.
{from: "SignupState"} "New user failure" "User email not verified!" E8: User email not verified! is displayed in the console after the user clicking on the "Let's start" button

Checked on the QA 1.23.1-2
Device: Desktop // Windows 10 // Chrome 88.0.4324.150

Attachment:
Video: https://www.dropbox.com/s/9m8vctxdz4scpr6/The%20issue%20with%20sign%20up%20with%20passwordless.mp4?dl=0

**Pull request merged**